### PR TITLE
Refactor rpc handlers and tests

### LINKED ISSUE #1750: Juno RPC handler code tidying
- Refactor the RPC handler code in Juno 
  - improve breakdown into code files, etc. 
  - refer to @kirugan for more suggestions re refactoring
- Increase coverage of automated tests
- Implement a pattern of generating RPC handler code scaffolding from OpenAPI spec

### PR DESCRIPTION
**Pull request by pnowosie**
Fixes #1750 

All handlers and related helper methods was moved from `rpc/handlers` to the corresponding "category" file, e.g. block, transaction, trace...

Handlers tests were refactor the same way.

I introduced new files in `rpc/` folder: 

- `chain.go` - with `ChainId` handler
- `class.go` - `Class*` handlers
-  `contract.go` - `Nonce` & `StorageAt`
- `estimate_fee.go`
- `rpc/helpers.go` which contains helper functionality used in more than one "category".

### CODE REVIEW

File: rpc/chain.go [hunk 1]
```diff
@@ -0,0 +1,324 @@
...
+	}
+
+	return rpcClass, nil
+}
+
+// ClassAt gets the contract class definition in the given block instantiated by the given contract address
+//
+// It follows the specification defined here:
+// https://github.com/starkware-libs/starknet-specs/blob/master/api/starknet_api_openrpc.json#L329
+func (h *Handler) ClassAt(id BlockID, address felt.Felt) (*Class, *jsonrpc.Error) {
```
Comment (rianhughes) [inline rpc/chain.go:157]: Can move to class.go?

File: rpc/chain.go [hunk 2]
```diff
@@ -0,0 +1,324 @@
...
+	classHash, err := h.ClassHashAt(id, address)
+	if err != nil {
+		return nil, err
+	}
+	return h.Class(id, *classHash)
+}
+
+// https://github.com/starkware-libs/starknet-specs/blob/e0b76ed0d8d8eba405e182371f9edac8b2bcbc5a/api/starknet_api_openrpc.json#L401-L445
+func (h *Handler) Call(funcCall FunctionCall, id BlockID) ([]*felt.Felt, *jsonrpc.Error) { //nolint:gocritic
```
Comment (rianhughes) [inline rpc/chain.go:166]: Create a call.go? Or move to trace.go??

File: rpc/chain.go [hunk 3]
```diff
@@ -0,0 +1,324 @@
...
+	if err != nil {
+		return nil, err
+	}
+
+	return utils.Map(result, func(tx SimulatedTransaction) FeeEstimate {
+		return tx.FeeEstimation
+	}), nil
+}
+
+func (h *Handler) EstimateMessageFee(msg MsgFromL1, id BlockID) (*FeeEstimate, *jsonrpc.Error) { //nolint:gocritic
```
Comment (rianhughes) [inline rpc/chain.go:240]: Move to its own file?

File: rpc/chain.go [hunk 4]
```diff
@@ -0,0 +1,324 @@
...
+	RevertError string `json:"revert_error"`
+}
+
+func makeContractError(err error) *jsonrpc.Error {
+	return ErrContractError.CloneWithData(ContractErrorData{
+		RevertError: err.Error(),
+	})
+}
+
+func (h *Handler) stateByBlockID(id *BlockID) (core.StateReader, blockchain.StateCloser, *jsonrpc.Error) {
```
Comment (rianhughes) [inline rpc/chain.go:302]: Move to state_update? Or create own file?

Comment (pnowosie) [reply rpc/chain.go:302]: It's used in many methods in chain / new class.go and one in simulate.go.
Maybe I'll  move it to helpers 🤔

File: rpc/chain.go [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+// https://github.com/starkware-libs/starknet-specs/blob/a789ccc3432c57777beceaa53a34a7ae2f25fda0/api/starknet_api_openrpc.json#L542
+func (h *Handler) ChainID() (*felt.Felt, *jsonrpc.Error) {
+	return h.bcReader.Network().L2ChainIDFelt(), nil
+}
+
+// StorageAt gets the value of the storage at the given address and key.
+//
+// It follows the specification defined here:
+// https://github.com/starkware-libs/starknet-specs/blob/a789ccc3432c57777beceaa53a34a7ae2f25fda0/api/starknet_api_openrpc.json#L110
+func (h *Handler) StorageAt(address, key felt.Felt, id BlockID) (*felt.Felt, *jsonrpc.Error) {
```
Comment (rianhughes) [inline rpc/chain.go:24]: What do you think about putting both Nonce and StorageAt in contract.go?

**Review by rianhughes — APPROVED** [review]

File: rpc/events.go [hunk 6]
```diff
@@ -30,3 +38,178 @@ type EmittedEvent struct {
...
+	h.mu.Lock()
+	sub, ok := h.subscriptions[id]
+	h.mu.Unlock() // Don't defer since h.unsubscribe acquires the lock.
+	if !ok || !sub.conn.Equal(w) {
+		return false, ErrSubscriptionNotFound
+	}
+	sub.cancel()
+	sub.wg.Wait() // Let the subscription finish before responding.
+	return true, nil
+}
```
Comment (kirugan) [inline rpc/events.go:113]: I'm not sure that `SubscribeNewHeads` and `Unsubscribe` related to events

**Review by kirugan — APPROVED** [review]

File: rpc/events.go (hunk 6 above)
Comment (pnowosie) [reply rpc/events.go:113]: We conclude with @kirugan I can merge this PR

**Pull request merged**